File: 203 Remove Linked List Elements/removeLinkedListElements.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeElements(struct ListNode* head, int val) {
    struct ListNode *prev = NULL;
    struct ListNode *cur = NULL;
    
    while (head && head->val == val) {
        head = head->next;
    }
    
    prev = head;
    while (prev) {
        cur = prev->next;
        if (cur && cur->val == val) {
            prev->next = cur->next;
        } else {
            prev = cur;
        }
    }
    
    return head;
}

File: 203 Remove Linked List Elements/removeLinkedListElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @param {integer} val
    # @return {ListNode}
    def removeElements(self, head, val):
        while head and head.val == val:
            head = head.next
            
        prev = head
        while prev:
            cur = prev.next
            if cur and cur.val == val:
                prev.next = cur.next
            else:
                prev = cur
                
        return head